examples/spec/query_spec.rb

```diff
@@ -0,0 +1,84 @@
+require_relative 'spec_helper'
+require_relative '../lib/query'
+
+RSpec.describe Query do
+  include RSpec::Hive::WithHiveConnection
+  include RSpec::Hive::QueryBuilderHelper
+
+  subject { described_class.new }
+
+  let(:schema) { subject.table_schema }
+
+  before { connection.execute(schema.create_table_statement) }
+
+  context 'without stubbing strategy' do
+    let(:input_data) do
+      [
+        ['Mikolaj', 'Cos', 1.23, 1],
+        ['Wojtek', 'Cos', 3.76, 2]
+      ]
+    end
+    let(:query_result) { connection.fetch(query) }
+
+    before { into_hive(schema).insert(*input_data).execute }
+
+    context 'when querying for amount > 3.2' do
+      let(:query) { "SELECT * FROM `#{subject.table_name}` WHERE amount > 3.2" }
+      let(:expected_result_array) do
+        [
+          [a_string_matching('Wojtek'), 'Cos', be_within(0.01).of(3.76)]
+        ]
+      end
+
+      it 'returns Wojtek' do
+        expect(query_result).to match_result_set(expected_result_array)
+      end
+    end
+
+    context 'when querying for amount < 3.2' do
+      let(:query) { "SELECT * FROM `#{subject.table_name}` WHERE amount < 3.2" }
+      let(:expected_result_hash) do
+        [
+          {name: 'Mikolaj', address: 'Cos', amount: be_within(0.01).of(1.23) }
+        ]
+      end
+
+      it 'returns Mikolaj' do
+        expect(query_result).to match_result_set(expected_result_hash)
+      end
+    end
+  end
+
+  context 'with stubbing strategy' do
```

Comment:
When using stubbing strategy (`into_hive(schema).insert(*input_data).with_stubbing.execute`). The values for columns that the user did not provide are randomly generated. As you can see in the `input_data` we only provide the name values for the columns. But once we get the result, the row does have values for address and amount as well.

This is supposed to be a showcase test, and there is a bit much going on...
1. you can give input values as an array of arrays or an array of hashes.
2. you can use composable matchers without our `match_result_set` matcher
3. for expected hashes, you can check for partial matches.

hm, maybe these concepts could be presented more clearly...